.codegen/api.go.tmpl

```diff
@@ -160,74 +161,86 @@ func (a *{{.Service.Name}}API) {{.PascalName}}AndWait(ctx context.Context{{if .R
 }
 {{end}}{{if .Pagination}}
 {{.Comment "// " 80}}
-// 
-// This method is generated by Databricks SDK Code Generator.
-func (a *{{.Service.Name}}API) {{.PascalName}}All(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) ([]{{.Pagination.Entity.PascalName}}, error) {
-	{{if .Pagination.MultiRequest}}var results []{{.Pagination.Entity.PascalName}}
-	{{ if .Pagination.Limit -}}
-	var totalCount {{template "type" .Pagination.Limit.Entity}} = 0
-	{{ end -}}
-	ctx = useragent.InContext(ctx, "sdk-feature", "pagination")
+func (a *{{.Service.Name}}API) {{.PascalName}}Iter({{if .Request}}request {{.Request.PascalName}}{{end}}) func (context.Context) (*{{.Pagination.Entity.PascalName}}, error) {
+	var err error
+	var isLastPage = false
+	var results []{{.Pagination.Entity.PascalName}}
 	{{if .Pagination.NeedsOffsetDedupe -}}
 	// deduplicate items that may have been added during iteration
 	seen := map[{{template "type" .Pagination.Entity.IdentifierField.Entity}}]bool{}
 	{{end}}{{if eq .Pagination.Increment 1 -}}
 	request.{{.Pagination.Offset.PascalName}} = 1 // start iterating from the first page
-	{{end}}for {
-		response, err := a.impl.{{.PascalName}}(ctx{{if .Request}}, request{{end}})
+	{{end}}return func(ctx context.Context) (*{{.Pagination.Entity.PascalName}}, error) {
```

Comment:
I wonder how would it look like if we port Java SDK's iteration implementation, moving the logic to a common class?

https://github.com/databricks/databricks-sdk-java/blob/main/databricks-sdk-java/src/main/java/com/databricks/sdk/support/Paginator.java#L14